src/mrsimulator/tests/test_core.py

```diff
@@ -92,6 +92,10 @@ def test_direct_init_isotopomer():
     assert id(the_isotopomer.sites[0] != the_isotopomer.sites[1])
     assert the_isotopomer.abundance == 10.0
 
+    # This should raise an error, but it is not.
+    the_isotopomer.sites[0] = "Trash"
+    assert the_isotopomer.sites[0] == "Trash"
```

Comment:
This should not be allowed. @shyamd Do you have any suggestion on how to set up traps for this?

Reply:
You're being too strict with python. That's perfectly fine behavior as far as python is concerned. It's very common practice to ask forgiveness rather than ask for permission. When you need something to act as a Site and not throw an error, just ensure it's an instance of a Site. Otherwise, just use it as a site and if it throws an error, that's ok.